Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/ConceptMergeController.java

```diff
@@ -0,0 +1,87 @@
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.servlet.ModelAndView;
+
+import edu.asu.conceptpower.app.bean.ConceptsMergeBean;
+import edu.asu.conceptpower.app.core.IConceptListManager;
+import edu.asu.conceptpower.app.core.IConceptManager;
+import edu.asu.conceptpower.app.core.IConceptTypeManger;
+import edu.asu.conceptpower.app.core.POS;
+import edu.asu.conceptpower.app.exceptions.DictionaryDoesNotExistException;
+import edu.asu.conceptpower.app.exceptions.DictionaryModifyException;
+import edu.asu.conceptpower.app.exceptions.IndexerRunningException;
+import edu.asu.conceptpower.app.exceptions.LuceneException;
+import edu.asu.conceptpower.app.service.IConceptMergeService;
+import edu.asu.conceptpower.core.ConceptEntry;
+import edu.asu.conceptpower.core.ConceptList;
+
+@Controller
+public class ConceptMergeController {
+
+    @Autowired
+    private IConceptManager conceptManager;
+
+    @Autowired
+    private IConceptMergeService conceptMergeService;
+
+    @Autowired
+    private IConceptTypeManger conceptTypesManager;
+
+    @Autowired
+    private IConceptListManager conceptListManager;
+
+    @Autowired
+    private ConceptsMergeBeanValidator validator;
+
+    @InitBinder
+    private void initBinder(WebDataBinder binder) {
+        binder.setValidator(validator);
+    }
+
+    @RequestMapping(value = "auth/concepts/merge", method = RequestMethod.GET)
+    public ModelAndView prepareMergeConcept(@ModelAttribute("conceptsMergeBean") ConceptsMergeBean conceptsMergeBean,
+            BindingResult result) {
+        List<ConceptEntry> conceptEntries = new ArrayList<>();
+        for (String id : conceptsMergeBean.getConceptIds()) {
+            conceptEntries.add(conceptManager.getConceptEntry(id));
+        }
+        conceptsMergeBean = conceptMergeService.prepareMergeConcepts(conceptEntries, conceptsMergeBean);
+        ModelAndView mav = new ModelAndView();
+        mav.addObject("types", conceptTypesManager.getAllTypes());
+        mav.addObject("conceptListValues", conceptListManager.getAllConceptLists().stream()
+                .map(ConceptList::getConceptListName).collect(Collectors.toSet()));
+        mav.addObject("posValues", POS.posValues);
+        mav.setViewName("/auth/conceptMerge");
+        return mav;
+    }
+
+    @RequestMapping(value = "auth/concepts/merge", method = RequestMethod.POST)
+    public ModelAndView mergeConcept(
+            @ModelAttribute("conceptsMergeBean") @Validated ConceptsMergeBean conceptsMergeBean, BindingResult result,
+            Principal principal) throws IllegalAccessException, LuceneException, IndexerRunningException,
+                    DictionaryDoesNotExistException, DictionaryModifyException {
+        ModelAndView mav = new ModelAndView();
+        if (result.hasErrors()) {
+            mav.setViewName("forward:/auth/conceptMerge");
```

Comment:
Yes this did not work. I set the view to the merge screen and loaded the drop down values alone. I did not redirect to the get method, because prepareMerge does some processing before directing to the view component. So I just set the value on tiles and returned the model.